### ISSUE TITLE
Search bar does not always start with selected contents

### ISSUE THREAD

**Issue by attah** [linuxmint/xed#402]
```
 * Xed version (xed --version) 2.6.2
 * Distribution - Mint 20
```

**Issue**
Search bar is unnecessarily stateful, and confusing.
Press ctrl-F at almost any time, and you have the contents of the search bar (if any)  selected and ready to be pasted over.
This is the same behaviour as most applications, nice and consistent.

However, if you have some text selected in the editor, it automatically gets copied to the search bar, _but it is not selected_ when the search bar is focused.
Select the same text again, or press ctrl-F again in genereal, and it does get selected. Very confusing if you are not paying super close attention.

**Steps to reproduce**
Select some text in the editor, press ctrl-F.

**Expected behaviour**
Search bar opens and its contents is selected.
(i have no preference in whether the contents is my selected text or not)

**Other information**
This appeared with the update to Mint 20, but has been the sorry state of Gedit since long ago (but not always).

**Comment by JosephMcc** [comment]
This is an annoying behavior. I looked into some last night but couldn't find a solution. I'll try to look into it some more later.

**Comment by jm4R** [comment]
Upvoted, very annoying behavior!

**Comment by abebinder** [comment]
Attempted a fix in https://github.com/linuxmint/xed/pull/722 and put all my research for why the problem happens there. 

TLDR:

async callback runs after the searchbar text is changed which changes selection to the first match in the document

https://github.com/linuxmint/xed/blob/78ff0d1db37872dcb3b236dc28e640f2e1f86720/xed/xed-searchbar.c#L140-L141

I tried to stop the callback from running for case where we select text in the document then do a search.
